Code/GraphMol/FileParsers/ProximityBonds.cpp

```diff
@@ -215,6 +215,27 @@ static void ConnectTheDots_Large(RWMol *mol, unsigned int flags) {
     unsigned int elem = atom->getAtomicNum();
     // detect multivalent Hs, which could happen with ConnectTheDots
     if (elem == 1 && atom->getDegree() > 1) {
+      // if there's an H neighbor and a non-H neighbor, remove the bond to the H
+      if (flags & ctdQUICKREMOVE_H_H_CONTACTS) {
+        Bond *bondToH = nullptr;
+        Bond *bondToNonH = nullptr;
+        for (auto bond : mol->atomBonds(atom)) {
+          if (bond->getOtherAtom(atom)->getAtomicNum() == 1) {
+            bondToH = bond;
```

Comment:
Could there be more than 1?

Reply:
Yeah, and then it falls through to the logic below